### PR TITLE
Add eitherTFromMaybe

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Found this implementation inside `eminence` for building an EitherT from a Maybe.
Not sure if it really belongs here or can be made simpler with another abstraction.

### CODE REVIEW

File: src/P/EitherT.hs [hunk 1]
```diff
@@ -11,3 +13,6 @@ firstEitherT f = bimapEitherT f id
 
 secondEitherT :: (Functor m) => (a -> b) -> EitherT e m a -> EitherT e m b
 secondEitherT f = bimapEitherT id f
+
+eitherTFromMaybe :: Functor m => b -> m (Maybe a) -> EitherT b m a
+eitherTFromMaybe l r = EitherT $ maybe (Left l) pure <$> r
```
Comment (domdere) [inline src/P/EitherT.hs]: The implementation looks a bit noisy /supernit

``` Haskell
eitherTFromMaybe :: Monad m => b -> m (Maybe a) -> EitherT b m a
eitherTFromMaybe l r = lift r >>= maybe (left l) return
```

But if the `Functor -> Monad` change isnt too palatable then nvm me

**Comment by domdere** [comment]
:+1:

**Comment by markhibberd** [comment]
:shaved_ice:

File: src/P/EitherT.hs (hunk 1 above)
Comment (tmcgilchrist) [inline src/P/EitherT.hs]: Pulling in `lift` would require more dependencies in P, namely adding `transformers` to the short list of dependencies in `p.cabal` Not sure if that's ok?

Comment (domdere) [inline src/P/EitherT.hs]: `transformers` is already a dependency on `either` so `p` already depends on it, but nvm, i think this is already too much thinking on the issue, its no longer worth the trouble, the code is fine :)

**Pull request merged**